### PR TITLE
add transistorProps

### LINKED ISSUE #126: Create prop for transistor
### Description:

- add transistorProps

### PR DESCRIPTION
**Pull request by Anshgrover23**
### Description:

- add transistorProps

Fixes #126

### CODE REVIEW

File: lib/components/transistor.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+} from "lib/common/layout"
+import { expectTypesMatch } from "lib/typecheck"
+import { z } from "zod"
+
+export interface TransistorProps extends CommonComponentProps {
+  transistorType: "NPN" | "PNP"
+  pinLabels: Record<"emitter" | "base" | "collector", string>
+  vce?: number | string
+  vbe?: number | string
+  ic?: number | string
```
Comment (seveibar) [inline lib/components/transistor.ts:14]: We try not to abbreviate unless the abbreviation is extremely well known

File: lib/components/transistor.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+import {
+  type CommonComponentProps,
+  commonComponentProps,
+} from "lib/common/layout"
+import { expectTypesMatch } from "lib/typecheck"
+import { z } from "zod"
+
+export interface TransistorProps extends CommonComponentProps {
+  transistorType: "NPN" | "PNP"
+  pinLabels: Record<"emitter" | "base" | "collector", string>
```
Comment (seveibar) [inline lib/components/transistor.ts:11]: Remove, we dont do this normally

Comment (seveibar) [reply lib/components/transistor.ts:11]: (For symbols)

File: lib/components/transistor.ts (hunk 1 above)
Comment (Anshgrover23) [reply lib/components/transistor.ts]: done.

File: lib/components/transistor.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
+import { voltage, current } from "circuit-json"
+import {
+  type CommonComponentProps,
+  commonComponentProps,
+} from "lib/common/layout"
+import { expectTypesMatch } from "lib/typecheck"
+import { z } from "zod"
+
+export interface TransistorProps extends CommonComponentProps {
+  transistorType: "NPN" | "PNP"
```
Comment (seveibar) [inline lib/components/transistor.ts:10]:
```suggestion
  transistorType: "npn" | "pnp"
```

File: lib/components/transistor.ts (hunk 2 above)
Comment (seveibar) [reply lib/components/transistor.ts:11]:
```suggestion
```

File: lib/components/transistor.ts [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+export const transistorProps = commonComponentProps.extend({
+  transistorType: z.enum(["NPN", "PNP"]),
+  pinLabels: z.object({
+    emitter: z.string(),
+    base: z.string(),
+    collector: z.string(),
+  }),
+  collectorEmitterVoltage: voltage.optional(),
+  baseEmitterVoltage: voltage.optional(),
+  collectorCurrent: current.optional(),
```
Comment (seveibar) [inline lib/components/transistor.ts:26]:
```suggestion
```

**Review by seveibar — APPROVED** [review]

**Pull request merged**